src/MiddlewareListener.php

```diff
@@ -91,6 +90,8 @@ public function onDispatch(MvcEvent $event)
             }
         }
 
+        $event->setError('');
```

Comment:
Note: this still means that the `DispatchListener` will try to execute and fail. The patch is not yet done. Ideally, the `DispatchListener` should check whether there is a `$event->getResult()`, and bail out if there already is one.